pubsub/service.go

```diff
@@ -91,3 +91,23 @@ func (r *streamingPullRetryer) Retry(err error) (pause time.Duration, shouldRetr
 		return r.defaultRetryer.Retry(err)
 	}
 }
+
+type publishRetryer struct {
+	defaultRetryer gax.Retryer
+}
+
+func (r *publishRetryer) Retry(err error) (pause time.Duration, shouldRetry bool) {
+	s, ok := status.FromError(err)
+	if !ok {
+		return r.defaultRetryer.Retry(err)
+	}
+	switch s.Code() {
```

Comment:
nit: This can be simplified to `if s.Code == codes.Internal && strings.Contains(...)`